master/buildbot/www/resource.py

```diff
@@ -117,3 +119,19 @@ def render(self, request):
         redir = self.base_url + self.basepath
         request.redirect(redir)
         return redir
+
+
+class InternalResource(Resource):
+    """
+    Resource available only for authorized users if authorization is enabled.
+    """
+    def render(self, request):
+        from .oauth2 import OAuth2Auth
+        wwwconfig = self.master.www
+        userinfos = wwwconfig.getUserInfos(request)
+        auth = wwwconfig.auth
+        if 'anonymous' in userinfos and isinstance(auth, OAuth2Auth) and auth.autologin:
```

Comment:
I would rather say:
```
 if 'anonymous' in userinfos and auth and auth.autologin:
     return auth.do_autologin(self, request)
```
and then implement the details of the autologin in the auth plugin.